src/txdb-leveldb.cpp

```diff
@@ -24,7 +24,43 @@ using namespace std;
 using namespace boost;
 
 leveldb::DB *txdb; // global pointer for LevelDB object instance
-void AddCPIDBlockHash(const std::string& cpid, const uint256& blockhash);
+
+namespace {
+//!
+//! \brief Set the correct CPID from the block claim when the block index
+//! contains a zero CPID.
+//!
+//! There were reports of 0000 cpid in index where INVESTOR should have been.
+//!
+//! \param pindex Index of the block to repair.
+//!
+void RepairZeroCpidIndex(CBlockIndex* const pindex)
```

Comment:
Excellent.

Reply:
This was all @denravonska older fix--I just encapsulated it in a function to simply the main index load function :slightly_smiling_face: